### DISCUSSION TITLE
Can I query the contents of a subcollection?

### DISCUSSION THREAD

**Discussion by GlodenBoy** [alirezanet/Gridify#87] [Q&A]
My data structure looks like this :

` class Test 
{
  int Id;
  List<Test> Children;
}`

`List<Test> data = new();`

I want to query the data for id=3 in the parent class and id=3 in the children

there should be many query conditions in the actual project, which roughly means that the parent class should be queried at the same time as the data that meets the conditions of the subclass

**Comment by alirezanet** [comment]
Yes, this snippet should help,
```csharp
var gm = new GridifyMapper<Test>()
         .GenerateMappings()
         .AddMap("child", q => q.Children.Select(x => x.Id));

      var actual = data.ApplyFiltering("id = 3, child = 3", gm).ToList();
```

**Reply by GlodenBoy** [reply]
thanks,

but ，This is just an example condition, if it is an n* condition , do I need to map all the attributes of child?

**Reply by alirezanet** [reply]
I think in that case you have to define a separate name for each level of nesting... child1, child2. 
or we will stock in an infinite loop

**Reply by GlodenBoy** [reply]
> I think in that case you have to define a separate name for each level of nesting... child1, child2. or we will stock in an infinite loop

Well, it will be more cumbersome to write like this,  in the future, I don't know if I can transfer the parent condition to the child class, which will be much more convenient,

**Reply by alirezanet** [reply]
Hmm, maybe I misunderstood what you really want, can you demonstrate what you want using a LINQ query?

**Reply by GlodenBoy** [reply]
You understand no problem, in my project The actual query of the parent class is a bit much, I want to query the "data" condition in "children" as well,

> sorry, my English is not very good, maybe there is a problem with the bing translation that caused you to misunderstand,

**Comment by GlodenBoy** [comment]
![image](https://user-images.githubusercontent.com/2329528/198035527-bec501d2-83d6-419f-b8d1-bc9281bd0062.png)

**Reply by alirezanet** [reply]
I believe you're looking for a recursive search feature that unfortunately gridify doesn't have. because it is not possible with LINQ. 
but let me know if you managed to prepare a reproducible example that you could've solved with a normal LINQ query, I probably add it for the next version.

**Reply by GlodenBoy** [reply]
![image](https://user-images.githubusercontent.com/2329528/198038888-5bc89470-33cb-4478-9c10-c78a5ef2c793.png)

The screenshot above can be easily converted to linq,

I'm indeed looking for nested queries that pass the parent's query to a subset

**Reply by alirezanet** [reply]
well, this is not recursive!
ohh I see, 
Yes, you need to map all the properties that you wanna search in the child class,

**Reply by GlodenBoy** [reply]
> well, this is not recursive! ohh I see, Yes, you need to map all the properties that you wanna search in the child class,

Well, thank you very much

**Reply by alirezanet** [reply]
You're welcome, good luck

```csharp
List<Test> data = new();

var qb = new QueryBuilder<Test>()
			.AddMap("id",q => q.Id)
			.AddMap("child_Id" , q=>q.Children.Select(c => c.Id))
			.AddMap("child_state" , q=>q.Children.Select(c => c.state))
			.Build(data);


public class Test
{
	public int Id;
	public string state;
	public List<Test> Children; 
}
```
